### PR TITLE
[PUPIL-797] New pupil lesson svg backgrounds and colours

### PR DESCRIPTION
**Pull request by BTitterington**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitive files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

Started [svg background ticket](https://www.notion.so/oaknationalacademy/OakComponents-OakLessonLayout-SVG-Illustration-applied-to-desktop-page-backgrounds-03b172a946894d47b3c8077971816d37) and then realised then also realised it needed the [overview and review colour updates](https://www.notion.so/oaknationalacademy/OakComponents-Lesson-overview-Lesson-review-colour-updates-065f72d47fdb4b7a9d808b20a87bcb5a), so I'm doing them as one ticket.

This adds primary and secondary backgrounds to the lesson overview and review, as well as the svg backgrounds for lesson introduction, quizzes and video.

The layout now has a celebrate property that will show the svg background on the quiz pages. I'm happy to review the naming etc in the code review.

Video for Richard 😁 
svg backgrounds

## A link to the component in the deployment preview

https://deploy-preview-240--lively-meringue-8ebd43.netlify.app/?path=/story/components-organisms-pupil-oaklessonlayout--default

## Testing instructions

You might need to drag the controls up from the bottom to change the sections, phase and celebrate option (which only takes effect on quiz sections).

 - Moving between the section types should change the background colours and borders etc.
 - For Overview and Review the phase can be changed to show a primary and secondary colour scheme.
 - The celebrate option should display the svg background on the quiz sections.
 - On mobile the svg and backgrounds aren't visible.

### CODE REVIEW

File: src/components/organisms/pupil/OakLessonLayout/OakLessonLayout.tsx [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+];
+
+export type LessonSectionName = (typeof lessonSectionNames)[number];
+
+type Phase = "primary" | "secondary";
 
 export type OakLessonLayoutProps = {
   lessonSectionName: LessonSectionName;
+  phase: Phase;
+  celebrate: boolean;
```
Comment (BTitterington) [inline src/components/organisms/pupil/OakLessonLayout/OakLessonLayout.tsx:27]: Happy to call this something else other than `celebrate` if there are some suggestions. This is to toggle the svg on the quiz pages when the answer is correct.

File: src/components/organisms/pupil/OakLessonLayout/OakLessonLayout.tsx [hunk 2]
```diff
@@ -43,6 +65,8 @@ const StickyFooter = styled(OakBox)`
  */
 export const OakLessonLayout = ({
   lessonSectionName,
+  phase = "primary",
+  celebrate = false,
```
Comment (BTitterington) [inline src/components/organisms/pupil/OakLessonLayout/OakLessonLayout.tsx:69]: defaults to ensure backwards compatibility.

File: src/components/organisms/pupil/OakLessonTopNav/OakLessonTopNav.tsx [hunk 3]
```diff
@@ -11,7 +13,7 @@ import {
 import { OakRoundIcon } from "@/components/molecules";
 import { getBreakpoint } from "@/styles/utils/responsiveStyle";
 
-type LessonSectionName = "intro" | "starter-quiz" | "video" | "exit-quiz";
+type LessonTopNavSectionName = Omit<LessonSectionName, "overview" | "review">;
```
Comment (BTitterington) [inline src/components/organisms/pupil/OakLessonTopNav/OakLessonTopNav.tsx:16]: Storybook was complaining that the lesson section names weren't compatible, so I've used the same sections but omitted `overview` and `review` as that matches the previous definition and I assume the nav isn't displayed on those sections.

**Review by messmike — APPROVED** [review]

**Review by RBrownsett — APPROVED** [review]
Approved - thanks for the video! Tested across different views, matches the designs

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**